code/datums/world_topic.dm

```diff
@@ -97,6 +97,9 @@
 	permit_insecure = TRUE
 
 /datum/world_topic/comms_console/Run(list/input, addr)
+	if(insecure_key && !CONFIG_GET(flag/insecure_announce))
+		return
```

Comment:
note for reviewers: I did consider caching the config in a static variable to avoid the list lookups, but didn't consider the gains worth it. I'm happy to do that if it's preferred though.